callbacks.tex

```diff
@@ -257,24 +257,49 @@ \section{Functional Programming}\label{s:callbacks-functional}
 a true believer in functional programming would be saddened by this:
 
 \begin{minted}{js}
+// Create test array
+const test = [1,2,3]
+
 const impure = (values) => {
   for (let i in values) {
     values[i] += 1
   }
 }
+
+// Run function
+impure(test)
+
+// Original array has been modified
+console.log(`test = ${test}`)
+\end{minted}
+
+\begin{minted}{text}
+test = 2,3,4
 \end{minted}
 
 \noindent
 and would politely, even patiently, suggest that it be rewritten like this:
 
 \begin{minted}{js}
+const test = [1,2,3]
+
 const pure = (values) -> {
   result = []
   for (let v of values) {
     result.push(v + 1)
   }
   return result
 }
+
+// Rather than modify test, we create a new array for the results
+const newArray = pure(test)
+console.log(`newArray = ${newArray}$`)
```

Comment:
I think `:` would be a better choice than `=` for the console messages in these two example code blocks. This would further reduce the risk of confusion between the information messages and an assignment operation.